stablehlo/reference/InterpreterOps.cpp

```diff
@@ -90,6 +92,32 @@ LogicalResult RunParallelOp::verify() {
         ") should match the sum of the number of results of all programs (",
         numResults, ")");
 
+  if (getInfeed()) {
+    if (getInfeed()->empty())
+      return emitOptionalError(
+          getLoc(), "infeed attribute is optional or should not be empty");
+
+    for (auto it = getInfeed()->begin(); it != getInfeed()->end(); ++it) {
+      auto funcName = (*it).dyn_cast<FlatSymbolRefAttr>().getAttr();
+      auto func =
+          (*this)->getParentOfType<ModuleOp>().lookupSymbol<func::FuncOp>(
+              funcName);
+      if (!func)
+        return emitOptionalError(getLoc(), "Function ", funcName, " not found");
+
+      if (func.getNumResults() != 1)
+        return emitOptionalError(getLoc(), "Function ", funcName,
+                                 " should return 1 tensor but returns ",
+                                 func.getNumResults());
+
+      for (auto type : func.getResultTypes())
+        if (!type.isa<ShapedType>())
```

Comment:
Probably don't need a loop if we've verified that there is only one result?

Can use something like:

```
auto resultType = func.getResultTypes()[0];
if (!resultType.isa<ShapedType>()) error(..., resultType);
```